app/src/test/java/com/gholem/moneylab/features/add/viewmodel/AddTransactionViewModelTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.gholem.moneylab.features.add.viewmodel
+
+import app.cash.turbine.test
+import com.gholem.moneylab.MainCoroutineRule
+import com.gholem.moneylab.arch.nav.NavigationLiveData
+import com.gholem.moneylab.common.BottomNavigationVisibilityBus
+import com.gholem.moneylab.domain.model.Transaction
+import com.gholem.moneylab.domain.model.TransactionCategory
+import com.gholem.moneylab.features.add.domain.InsertTransactionsModelUseCase
+import com.gholem.moneylab.features.add.navigation.AddNavigationEvent
+import com.gholem.moneylab.features.chooseTransactionCategory.domain.GetCategoryListUseCase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.*
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class AddTransactionViewModelTest {
+
+    @get:Rule
+    val mainCoroutineRule = MainCoroutineRule()
+
+    private val insertTransactionsModelUseCaseMock: InsertTransactionsModelUseCase =
+        mock(InsertTransactionsModelUseCase::class.java)
+    private val bottomNavigationVisibilityBusMock: BottomNavigationVisibilityBus =
+        mock(BottomNavigationVisibilityBus::class.java)
+    private val getCategoryListUseCaseMock: GetCategoryListUseCase =
+        mock(GetCategoryListUseCase::class.java)
+    private val navigationMock: NavigationLiveData<AddNavigationEvent> =
+        mock(NavigationLiveData::class.java) as NavigationLiveData<AddNavigationEvent>
+
+    private lateinit var viewModel: AddTransactionViewModel
+
+    @Before
+    fun setup() {
+        viewModel = AddTransactionViewModel(
+            insertTransactionsModelUseCaseMock,
+            bottomNavigationVisibilityBusMock,
+            getCategoryListUseCaseMock
+        )
+        viewModel.navigation = navigationMock
+    }
+
+    @Test
+    fun `verify invocations on navigateToCategoryBottomSheet method call`() = runTest {
+        /* When */
+        viewModel.navigateToCategoryBottomSheet()
+
+        /* Then */
+        verify(viewModel.navigation).emit(AddNavigationEvent.ToCategoryBottomSheetDialog)
```

Comment:
Instead of mocking the navigation and using `verify` you can simply remove the mock (and also change navigation in all view model from `var` to `val`) and check it in tests like:

assertEquals(AddNavigationEvent.ToCategoryBottomSheetDialog, viewModel.navigation.value?.getAndForget())